hbase-server/src/main/java/org/apache/hadoop/hbase/regionserver/StoreScanner.java

```diff
@@ -351,11 +350,11 @@ public StoreScanner(ScanInfo scanInfo, ScanType scanType,
 
   // Used to instantiate a scanner for compaction in test
   @VisibleForTesting
-  StoreScanner(ScanInfo scanInfo, OptionalInt maxVersions, ScanType scanType,
+  StoreScanner(ScanInfo scanInfo, int maxVersions, ScanType scanType,
       List<? extends KeyValueScanner> scanners) throws IOException {
     // 0 is passed as readpoint because the test bypasses Store
-    this(null, maxVersions.isPresent() ? new Scan().readVersions(maxVersions.getAsInt())
-        : SCAN_FOR_COMPACTION, scanInfo, 0, 0L, false, scanType);
+    this(null, maxVersions > 0 ? new Scan().readVersions(maxVersions)
```

Comment:
To be honest I'm not sure if > 0 is a valid condition... Need to read the code... Better keep it as is and open another issue for this...

Reply:
From the code, I felt > 0 could be safe bet(this is maxVersion of Scan, default value is 1), btw this constructor is not being used anywhere else other than unit tests: TestStoreScanner.java
Hence, is it fine to have this check?

Reply:
Fine.